crates/criticalup-core/src/download_server_client.rs

```diff
@@ -117,17 +118,28 @@ impl DownloadServerClient {
         // If the token contains such chars treat the authentication as failed due to an invalid
         // token, as the server wouldn't be able to validate it either anyway.
 
-        // set path to token file for docker
+        // Set path to token file for docker.
         let path_to_token_file = if std::path::Path::new("/.dockerenv").exists() {
             Some("/run/secrets/CRITICALUP_TOKEN")
         } else {
             None
         };
 
-        let header = self
+        // Check for token in
+        // 1. Docker
+        // 2. State file (included in call to `self.state.authentication_token(...)`)
+        // 3. Environment Variable
+        let searched_token = self
             .state
             .authentication_token(path_to_token_file)
```

Comment:
Seems having `.state` read some passed token file isn't maybe what we want, right? Why not read the token file and skip checking state instead?

In my head I imagine something (obviously boiled down):

```rust
let token_from_file: Option<String> = if std::path::Path::new("/.dockerenv").exists() {
   read_to_string("/run/secrets/CRITICALUP_TOKEN").ok()
};
let token_from_env: Option<String> = os::env::var("CRITICALUP_TOKEN").ok();
let token_from_state: Option<String> = self.state.authentication_token();

let token = match (token_from_file, token_from_env, token_from_state) {
  (Some(token), _, _) => token,
  (None, Some(token), _) => token,
  (None, None, Some(token) => token,
}
```